dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/client/ConfigProviderResolverImpl.java

```diff
@@ -0,0 +1,91 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.restfulWS.client;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigBuilder;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+/**
+ * Only intended to be used in Java SE client environments.
+ */
+public class ConfigProviderResolverImpl extends ConfigProviderResolver {
+    
+    private final boolean java2SecurityEnabled = System.getSecurityManager() != null;
+
+    private final Config config = new Config() {
+
+        @Override
+        public Iterable<ConfigSource> getConfigSources() {
+            // TODO Auto-generated method stub
+            return null;
+        }
+
+        @Override
+        public <T> Optional<T> getOptionalValue(String propName, Class<T> type) {
+            return Optional.ofNullable(getValue(propName, type));
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Iterable<String> getPropertyNames() {
+            return (Iterable<String>) System.getProperties().keySet().iterator();
```

Comment:
This is more of a question. I know this is an SE environment only, but do we need do-privs when getting system properties? If this wasn't SE would we need them?

Reply:
Good catch.  Our FATs and the TCK most likely wouldn't test with a Java 2 security manager in a Java SE environment (and this code should never run outside of a Java SE env), but it is good practice to wrap it in a doPriv block.  I'll update.